src/globus_compute_common/messagepack/message_types/result.py

```diff
@@ -16,10 +16,23 @@ class ResultErrorDetails(pydantic.BaseModel):
     user_message: str
 
 
+@meta(message_type="result")
+class ResultV1(Message):
+    task_id: uuid.UUID
+    data: str
+    error_details: t.Optional[ResultErrorDetails]
+    task_statuses: t.Optional[t.List[TaskTransition]]
+
+    @property
+    def is_error(self) -> bool:
+        return self.error_details is not None
+
+
 @meta(message_type="result")
 class Result(Message):
     task_id: uuid.UUID
     data: str
+    details: t.Optional[t.Dict[t.Any, t.Any]]
     error_details: t.Optional[ResultErrorDetails]
     task_statuses: t.Optional[t.List[TaskTransition]]
```

Comment:
If this is a strict addition, do we need to hold on to `ResultV1` in this case?  We aren't removing or altering a field.  I _think_ messagepack handles this style of addition gracefully, given that it's a schema-less format.

Reply:
Yeah, I think the test of whether we even need `ResultV1` will just be to show that the Result Processor utilizing compute-common from before this PR does not bork processing packets from an endpoint utilizing this updated `Result` format.  If that's successful, then testing should be a "simple" matter of just verifying the new field.

Reply:
As discussed, moved ResultV1 to test but left it there as I wasn't as certain of the ser/de internals thus making sure.